DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/ManageCourseControllerTests.cs

```diff
@@ -109,11 +155,155 @@ public void
+            A.CallTo(() => courseService.GetCourseOptionsAlphabeticalListForCentre(A<int>._, A<int>._))
+                .Returns(courseOptions);
+
+            A.CallTo(
+                () => courseService.UpdateLearningPathwayDefaultsForCourse(
+                    A<int>._,
+                    A<int>._,
+                    A<int>._,
+                    A<bool>._,
+                    A<bool>._,
+                    A<int>._,
+                    A<int>._,
+                    A<bool>._
+                )
+            ).DoesNothing();
+
+            // When
+            var result = controller.SaveAutoRefreshOptions(1, autoRefreshModel);
+
+            // Then
+            A.CallTo(
+                () => courseService.UpdateLearningPathwayDefaultsForCourse(
+                    1,
+                    6,
+                    12,
+                    false,
+                    false,
+                    1,
+                    0,
+                    true
+                )
+            ).MustHaveHappened();
+            result.Should().BeRedirectToActionResult().WithActionName("Index");
+        }
+
+        [Test]
+        public void SaveAutoRefreshOptions_does_not_call_service_with_invalid_model()
+        {
+            // Given
+            var learningPathwayModel = new EditLearningPathwayDefaultsViewModel(1, "6", "12", false, false);
+            var autoRefreshModel = new EditAutoRefreshOptionsViewModel(1, 1, "13", true);
+            controller.ModelState.AddModelError("AutoRefreshMonths", "Enter a whole number from 0 to 12");
+
+            var learningPathwayData = new EditLearningPathwayDefaultsData(learningPathwayModel);
+            controller.TempData.Set(learningPathwayData);
+
+            var courseOptions = new List<(int, string)> { (1, "courseName") };
+            A.CallTo(() => courseService.GetCourseOptionsAlphabeticalListForCentre(A<int>._, A<int>._))
+                .Returns(courseOptions);
+
+            A.CallTo(
+                () => courseService.UpdateLearningPathwayDefaultsForCourse(
+                    A<int>._,
+                    A<int>._,
+                    A<int>._,
+                    A<bool>._,
+                    A<bool>._,
+                    A<int>._,
+                    A<int>._,
+                    A<bool>._
+                )
+            ).DoesNothing();
+            controller.ModelState.AddModelError("RefreshToCustomisationId", "Select a course");
+
+            // When
+            var result = controller.SaveAutoRefreshOptions(1, autoRefreshModel);
+
+            // Then
+            A.CallTo(
+                () => courseService.UpdateLearningPathwayDefaultsForCourse(
+                    1,
+                    6,
+                    12,
+                    false,
+                    false,
+                    1,
+                    13,
+                    false
+                )
+            ).MustNotHaveHappened();

```

Comment:
We should not explicitly set the values in here as we want to be sure that no calls to the service have happened. This test would still pass if e.g. courseService.UpdateLearningPathwayDefaultsForCourse(2, .....) happened

Reply:
This is also true of some other tests in this file